javascript/javascript3/week2/currency-caculator.js

```diff
@@ -0,0 +1,60 @@
+const amountTag = document.getElementById("amount");
+const fromTag = document.getElementById("from");
+const toTag = document.getElementById("to");
+const resultTag = document.getElementById("result");
+const btnTag = document.getElementById("converted-amount");
+
+btnTag.addEventListener("click", convertCurrency);
+
+let exchangeRates = {};
+let fromCurrency = "EUR";
+let toCurrency = "DKK";
+
+async function getDataAndConvertFromApi() {
+	try {
+		const response = await fetch("https://open.er-api.com/v6/latest/USD");
+		const data = await response.json();
+		exchangeRates = data.rates;
+
+		const currencies = Object.keys(exchangeRates);
+
+		const fromOptionVal = document.createElement("option");
+		fromOptionVal.text = "EUR";
+		fromOptionVal.value = "EUR";
+		fromTag.appendChild(fromOptionVal);
+
+		const toOptionVal = document.createElement("option");
+		toOptionVal.text = "DKK";
+		toOptionVal.value = "DKK";
+		toTag.appendChild(toOptionVal);
+
+		currencies.forEach((currency) => {
+			const option1 = document.createElement("option");
+			option1.text = currency;
+			fromTag.add(option1);
+
+			const option2 = document.createElement("option");
+			option2.text = currency;
+			toTag.add(option2);
+		});
```

Comment:
This is a huge function. Try to seperate it. Use an async function to fetch api, then another function to do the logic, etc

Reply:
ok.